File: array_problems/countTargetChar/countTargetChar.cs
using System;
using System.Threading.Tasks;
using System.Collections.Generic;

class ArrayProblems {
    public async Task<int> CountTargetChar(string str, char targetChar ) {
        List<char> result = new List<char>();
        str = str.ToLower();
        if(str.Length == 0) {
            return 0;
        }
        for(int i = 0; i < str.Length; i++) {
            if(str[i] == char.ToLower(targetChar)) {
                result.Add(str[i]);
            }
        }
        return result.Count;
    }
}

public class Program {
    public static async Task Main(string[] args) {
        ArrayProblems arrayProblems = new ArrayProblems();
        int numberChar = await arrayProblems.CountTargetChar("Hello world, this is very basic for developers", 'S');
        Console.WriteLine(numberChar);
    }
}

File: array_problems/countTargetChar/countTargetChar.go
package main
import (
    "fmt"
    "strings"
)
type ArrayProblems struct{}
func (ap *ArrayProblems) CountTargetChar(str string, targetChar rune) int {
    result := []int{}
    str = strings.ToUpper(string(str))
    if len(str) == 0 {
        return 0;
    }
    for _, ch := range str {
        if(ch == targetChar || ch == targetChar-'a'+'A') {
            result = append(result, int(ch))
        }
    }
    return len(result)
}

func main() {
    arrayProblems := ArrayProblems{}
    numberChar := arrayProblems.CountTargetChar("Hello world, this is very basic for developers", 'S')
    fmt.Println(numberChar)
}